marlowe/src/Language/Marlowe/Extended/V1/Metadata/Types.hs

```diff
@@ -0,0 +1,130 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Language.Marlowe.Extended.V1.Metadata.Types
+  where
+
+import Data.Aeson (FromJSON, Value, parseJSON, withArray, withObject, (.:))
+import Data.Aeson.Types (Parser)
+import Data.Bifunctor (Bifunctor(first))
+import qualified Data.Foldable as F
+import Data.Map (Map)
+import qualified Data.Map as Map
+import Data.Map.Ordered (OMap)
+import qualified Data.Map.Ordered as OMap
+import Data.Set (Set)
+import Data.Set.Ordered (OSet)
+import Data.Text.Encoding as Text
+import qualified GHC.Generics
+import Plutus.V1.Ledger.Api (TokenName)
+import qualified Plutus.V1.Ledger.Value as Val
+
+data ContractType
+  = Escrow
+  | EscrowWithCollateral
+  | ZeroCouponBond
+  | CouponBondGuaranteed
+  | Swap
+  | ContractForDifferences
+  | Other
```

Comment:
Would it make sense for `Other` to be `Other String` so that the "other" type of contract could be named?

Reply:
Totally... I've set it the same as it is in PureScript for compatibility purposes, but I'd like to get rid of this field altogether (or find better taxonomy as you mentioned later). But I'd like that to happen after we resolve this discussion:
https://github.com/input-output-hk/MIPs/discussions/7